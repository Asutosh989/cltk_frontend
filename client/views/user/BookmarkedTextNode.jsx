import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

BookmarkedTextNode = React.createClass({

	propTypes: {
		text: React.PropTypes.object.isRequired,
		isOdd: React.PropTypes.bool,
	},

	childContextTypes: {
		muiTheme: React.PropTypes.object.isRequired,
	},

	getChildContext() {
		return { muiTheme: getMuiTheme(baseTheme) };
	},

	mixins: [ReactMeteorData],

	getMeteorData() {
		let work = null;
		const query = { _id: this.props.text.work };
		const handleWorks = Meteor.subscribe('works', query);
		work = Works.findOne(query);

		return {
			work,
		};
	},

	getTextLocation() {
		const text = this.props.text;
		let location = '';
		let textN = '';

		if ('n_1' in text) {
			location += text.n_1;
			textN = text.n_1;
		}
		if ('n_2' in text) {
			location += `.${text.n_2}`;
			textN = text.n_2;
		}
		if ('n_3' in text) {
			location += `.${text.n_3}`;
			textN = text.n_3;
		}
		if ('n_4' in text) {
			location += `.${text.n_4}`;
			textN = text.n_4;
		}
		if ('n_5' in text) {
			location += `.${text.n_5}`;
			textN = text.n_5;
		}

		return {
			location,
			textN,
		};
	},

	handleClick() {
		debugger;

	},


	render() {
		const text = this.props.text;
		let textClasses = 'text-node bookmark-text-node clearfix';
		const textLocation = this.getTextLocation();
		let workTitle = '';
		let link = '';

		if (this.data.work) {
			workTitle = this.data.work.english_title;
			link = `/works/${this.data.work._id}/${this.data.work.slug}?location=${textLocation.location}`;
		}

		if (this.props.isOdd) {
			textClasses = `${textClasses} bookmark-text-node--odd`;
		}

		if ((parseInt(textLocation.textN, 10) % 5) === 0) {
			textClasses = `${textClasses} show-number`;
		}

		return (
			<a
				className={textClasses}
				data-id={text._id}
				data-loc={textLocation.location}
				href={link}
			>
				<div className="text-left-header">
					<h2 className="section-numbering">{Utils.trunc(workTitle, 40)} {textLocation.location}</h2>
				</div>

				<p
					className="text-html"
					ref={(ref) => {
						this.anchorEl = ref;
						return ref;
					}}
				>
					{text.text && text.text .length ?
						<span>{Utils.trunc(text.text, 120)}</span>
					:
						<span>[ . . . ]</span>
					}
				</p>

			</a>
		);
	},
});
