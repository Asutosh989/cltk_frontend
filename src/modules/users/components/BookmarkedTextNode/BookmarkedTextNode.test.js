import React from 'react';
import { shallow } from 'enzyme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import BookmarkedTextNode from './BookmarkedTextNode';

describe('BookmarkedTextNode', () => {
	it('renders correctly', () => {
		const wrapper = shallow(
			<MuiThemeProvider>
				<BookmarkedTextNode
					text={{}}
				/>
			</MuiThemeProvider>
		);
		expect(wrapper).toBeDefined();
	});
});
