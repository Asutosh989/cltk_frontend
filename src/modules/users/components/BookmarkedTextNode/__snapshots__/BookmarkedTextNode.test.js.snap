// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkedTextNode renders correctly 1`] = `
<a
  className="text-node bookmark-text-node clearfix"
  data-id={undefined}
  data-loc=""
  href=""
>
  <div
    className="text-left-header"
  >
    <h2
      className="section-numbering"
    >
      
       
      
    </h2>
  </div>
  <p
    className="text-html"
  >
    <span>
      [ . . . ]
    </span>
  </p>
</a>
`;
